### ISSUE TITLE
Feature request: Add option to show the result of the 3 way diff, i.e. the real patch that will be applied

### ISSUE THREAD

**Issue by max-rocket-internet** [helm/helm#8718]
It would be really great if Helm could print the generated patch from the [3 way diff](https://github.com/helm/helm/pull/6124/files).

This would allow us to see exactly what will be applied to the cluster. I know there's the `--dry-run` option for `helm upgrade` but this is just prints the whole rendered chart and it's very difficult to tell what has changed in complex charts.

Maybe something like this?

```diff
$ helm upgrade my-chart deliveryhero/hoppscotch --values=path/to/my/values.yaml --dry-run --diff

Comparing release=my-chart, chart=deliveryhero/hoppscotch
default, hoppscotch, Deployment (apps/v1) has changed:

  # Source: hoppscotch/templates/deployment.yaml
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: hoppscotch
    labels:
      app.kubernetes.io/name: hoppscotch
      helm.sh/chart: hoppscotch-0.2.7
      app.kubernetes.io/instance: hoppscotch
  spec:
-   replicas: 2
+   replicas: 6
    selector:
      matchLabels:
        app.kubernetes.io/name: hoppscotch
        app.kubernetes.io/instance: hoppscotch
    template:
      metadata:
        labels:
          app.kubernetes.io/name: hoppscotch
          app.kubernetes.io/instance: hoppscotch
      spec:
-       serviceAccountName: hoppscotch-default
+       serviceAccountName: my-custom-serviceaccount
        securityContext:
          {}
        containers:
          - name: hoppscotch
```

Or simply just print the patch as is like this:

```yaml
  # Source: hoppscotch/templates/deployment.yaml
  spec:
    replicas: 6
    template:
      spec:
        serviceAccountName: my-custom-serviceaccount
```

I know `helm-diff` plugin exists this is broken in Helm 3, background is in https://github.com/databus23/helm-diff/issues/176

**Comment by bacongobbler** [comment]
This seems like a useful idea for a plugin. The patch can be generated without helm’s assistance by using the same logic available in `pkg/kube/client.go`.

**Comment by max-rocket-internet** [comment]
@bacongobbler 

> This seems like a useful idea for a plugin.

We already have the plugin and it doesn't work with Helm 3. That's what https://github.com/databus23/helm-diff/issues/176 is about.

I think getting Helm to simply show what it's sending to the API sounds more like a debug option for Helm itself as opposed to a plugin? Something like `--show-patch`?

**Comment by bacongobbler** [comment]
Okay. I wrote something up in a few spare hours this morning. It's much more complex than a feature flag addition to Helm, which is why I suggested it should be a standalone plugin similar to `helm-diff`. This is just a proof of concept, but it generates the JSON diff that Helm generates internally before sending it to Kubernetes. Feel free to take a look.

https://github.com/bacongobbler/helm-patchdiff
